internal/fs/dir_handle.go

```diff
@@ -26,15 +28,30 @@ import (
 	"golang.org/x/net/context"
 )
 
+type Record struct {
+	sync.Mutex
+	length          int
+	err             error
+	entryForSorting fuseutil.Dirent
+	cond            *sync.Cond
+}
+
+func NewRecord() *Record {
+	r := Record{}
+	r.cond = sync.NewCond(&r)
+	return &r
+}
+
 var ContinuationToken string
+var rec = NewRecord()
 
 // State required for reading from directories.
-type dirHandle struct {
+type DirHandle struct {
```

Comment:
Why did you make it public ?

Reply:
For unit testing, I needed to export dirHandle from fs package

Reply:
In that case can't you use getter setter methods?

Reply:
Yes.Will update the code.